files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,11 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		// userID cannot be set during an AJAX request, but only when calling this method within PHP
```

Comment:
@dtdesign Hm, I don't really get it. What's difference between:
-  unsetting the userID in the `validateMarkAsRead` method (which handles AJAX requests) and setting in to the currents userID in the `markAsRead` method
- setting the userID to the currents userID in the `validateMarkAsRead` method

?